### PR TITLE
print to the log output when a workspace mapping fails

### PR DESCRIPTION
**Pull request by moswald**

### CODE REVIEW

**Comment by moswald** [comment]
@ericsciple There was a discussion on this a week ago or so, this is just following through with the change.

File: src/Agent.Worker/Build/TfsVCSourceProvider.cs [hunk 1]
```diff
@@ -679,15 +679,15 @@ public static class WorkspaceUtil
                         bool expectedCloak = definitionMapping.MappingType == DefinitionMappingType.Cloak;
                         if (tfMapping.Cloak != expectedCloak)
                         {
-                            executionContext.Debug($"Expected mapping[{i}] cloak: '{expectedCloak}'. Actual: '{tfMapping.Cloak}'");
+                            executionContext.Output($"Expected mapping[{i}] cloak: '{expectedCloak}'. Actual: '{tfMapping.Cloak}'");
```
Comment (TingluoHuang) [inline src/Agent.Worker/Build/TfsVCSourceProvider.cs]: you might want to localize all the strings, since it's now go to `output` instead of `debug`

Comment (TingluoHuang) [reply src/Agent.Worker/Build/TfsVCSourceProvider.cs]: `StringUtil.Loc()`

**Comment by TingluoHuang** [comment]
@moswald why only change those 3 `Debug()` to `Output()`?
Will this causing noisy log for success case? if you think those information are critical, you can buffer them with `StringBuilder` and print the result only when not match.
Double check what the output looks like for hosted build, i remembered we did something special for hosted build with TFVC.

File: src/Agent.Worker/Build/TfsVCSourceProvider.cs (hunk 1 above)
Comment (moswald) [reply src/Agent.Worker/Build/TfsVCSourceProvider.cs]: Good point, I didn't think of that.

**Comment by moswald** [comment]
@TingluoHuang The scope of the change was just for failed mappings for Tfvc workspaces. These will only show if a candidate workspace has matched all the other requirements (name, machine, mapping count, etc).

**Comment by TingluoHuang** [comment]
@moswald i am going to resolve the conflict and merge it for you. #1652
i am in the process of releasing agents, i will include this PR in this release.

**Comment by moswald** [comment]
@TingluoHuang Thanks, I completely forgot about this PR.

**Pull request closed without merging**